crates/gourgeist/src/bare.rs

```diff
@@ -226,31 +227,68 @@ pub fn create_bare_venv(
     } else {
         unimplemented!("Only Windows and Unix are supported")
     };
-    let pyvenv_cfg_data = &[
-        ("home", python_home),
-        (
-            "implementation",
-            interpreter.markers().platform_python_implementation.clone(),
-        ),
-        (
-            "version_info",
-            interpreter.markers().python_version.string.clone(),
-        ),
-        ("gourgeist", env!("CARGO_PKG_VERSION").to_string()),
-        // I wouldn't allow this option anyway
-        ("include-system-site-packages", "false".to_string()),
-        (
-            "base-prefix",
-            interpreter.base_prefix().to_string_lossy().to_string(),
-        ),
-        (
-            "base-exec-prefix",
-            interpreter.base_exec_prefix().to_string_lossy().to_string(),
-        ),
-        ("base-executable", base_python.to_string()),
+
+    // Validate extra_cfg
+    let reserved_keys = [
+        "home",
+        "implementation",
+        "version_info",
+        "gourgeist",
+        "include-system-site-packages",
+        "base-prefix",
+        "base-exec-prefix",
+        "base-executable",
+        "prompt",
     ];
+    for (key, _) in &extra_cfg {
+        if reserved_keys.contains(&key.as_str()) {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("Reserved key found in extra_cfg: {key}"),
+            ));
+        }
+    }
+
+    let pyvenv_cfg_data: Vec<(String, String)> = [
+        vec![
+            ("home".to_string(), python_home),
+            (
+                "implementation".to_string(),
+                interpreter.markers().platform_python_implementation.clone(),
+            ),
+            (
+                "version_info".to_string(),
+                interpreter.markers().python_version.string.clone(),
+            ),
+            (
+                "gourgeist".to_string(),
+                env!("CARGO_PKG_VERSION").to_string(),
+            ),
+        ],
+        // Put custom cfg pairs after "gourgeist"
+        extra_cfg,
+        vec![
+            (
+                "include-system-site-packages".to_string(),
+                "false".to_string(),
+            ),
+            (
+                "base-prefix".to_string(),
+                interpreter.base_prefix().to_string_lossy().to_string(),
+            ),
+            (
+                "base-exec-prefix".to_string(),
+                interpreter.base_exec_prefix().to_string_lossy().to_string(),
+            ),
+            ("base-executable".to_string(), base_python.to_string()),
+        ],
+    ]
```

Comment:
While your at it, would you mind moving `prompt` also to this list?

Reply:
Done in 258e4034a7aec9768ce0cee8121fafa3f44bf6b5

Reply:
thanks!